dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java

```diff
@@ -87,7 +87,9 @@ public URL getUrl() {
             return invokers;
         }
 
-        if (tagRouterRule == null || !tagRouterRule.isValid() || !tagRouterRule.isEnabled()) {
+        // since the rule can be changed by config center, we should dump one to use.
+        TagRouterRule dump = tagRouterRule;
```

Comment:
Shall we mark it as final? Also I don't quite understand why it is named as dump?
If it is just a copy, will a name called `tagRouterRuleCopy` be better?

Reply:
Agree, tagRouterRuleCopy or CopyOfXxx would be better.

Reply:
For me, the semantics of `copy` or `refer` are good choices. So, I'd like chose `copy` in this case.